packages/bsky/src/auth.ts

```diff
@@ -30,6 +30,45 @@ export const authOptionalVerifier =
     return authVerifier(idResolver, opts)(reqCtx)
   }
 
+export const authOptionalAccessOrRoleVerifier = (
+  idResolver: IdResolver,
+  cfg: ServerConfig,
+) => {
+  const verifyAccess = authVerifier(idResolver, { aud: cfg.serverDid })
+  const verifyRole = roleVerifier(cfg)
+  return async (ctx: { req: express.Request; res: express.Response }) => {
+    const defaultUnAuthorizedCredentials = { credentials: { did: null } }
```

Comment:
We could add a `type` to this so that all the resulting credentials (role, access, and unauthed) can be easily distinguished from each other.